source/extensions/filters/http/cache/http_cache.cc

```diff
@@ -188,48 +187,56 @@ std::vector<RawByteRange> RangeRequests::parseRanges(const Http::RequestHeaderMa
   Http::HeaderUtility::getAllOfHeader(request_headers, Http::Headers::get().Range.get(),
                                       range_headers);
 
-  absl::string_view range;
+  absl::string_view header_value;
   if (range_headers.size() == 1) {
-    range = range_headers.front();
+    header_value = range_headers.front();
   } else {
-    ENVOY_LOG(debug, "Multiple range headers provided in request. Ignoring all range headers.");
+    if (range_headers.size() > 1) {
+      ENVOY_LOG(debug, "Multiple range headers provided in request. Ignoring all range headers.");
+    }
     return {};
   }
 
-  if (!absl::ConsumePrefix(&range, "bytes=")) {
+  if (!absl::ConsumePrefix(&header_value, "bytes=")) {
     ENVOY_LOG(debug, "Invalid range header. range-unit not correctly specified, only 'bytes' "
                      "supported. Ignoring range header.");
     return {};
   }
 
-  std::vector<RawByteRange> ranges;
-  int byte_ranges_parsed = 0;
+  std::vector<absl::string_view> ranges = absl::StrSplit(header_value, ',');
```

Comment:
If we limit to a maximum of `byte_range_parse_limit` elements, we wouldn't know straightaway if there are more ranges than acceptable (see if statement on next line).
The function would still work fine (since we have a check if `cur_range` is empty after reading what we want), but we'd potentially parse `byte_range_parse_limit` ranges before aborting.

To prevent the start of the loop when we have too many ranges we could:
1. use a limit of `byte_range_parse_limit + 1`, so we still limit splits, but then we can find out if there are extra ranges just with `.size()`.
2. use a limit of `byte_range_parse_limit` then check if size is at the max, if it is, check if there's a comma in the last element.
Thoughts?

Reply:
Part of the problem is that byte_range_parse_limit is an ambiguous name. I suggest renaming it to max_byte_range_specs. 

MaxSplits's second argument specifies the max number of delimiters to match, not the max number of elements. Once it finds the nth delimiter, the rest of the string goes into element n+1.
Example:
StrSplit("1-2,3-4,5-6,7-8", MaxSplits(',', 1)) == ["1-2", "3-4,5-6,7-8"]
StrSplit("1-2,3-4,5-6,7-8", MaxSplits(',', 2)) == ["1-2", "3-4", "5-6,7-8"]

To check for too many 
If (ranges.size() > max_byte_range_specs) return {};

Reply:
Oh, okay. I got confused with the delimiter max value.
That makes sense, that's what I was suggesting on my 1st option (with my misunderstanding of the parameter).

I'll make these changes.